OpenChange/MAPIStoreMailMessage.m

```diff
@@ -312,7 +306,19 @@ - (void) _fetchHeaderData
           partParameters = [partHeaderData objectForKey: @"parameterList"];
           encoding = [partHeaderData objectForKey: @"encoding"];
           charset = [partParameters objectForKey: @"charset"];
-          mimeType = [[keys objectAtIndex: i] objectForKey: @"mimeType"];  
+          mimeType = [bodyStructureKey objectForKey: @"mimeType"];
+
+          /* multipart/mixed is the default type.
+             multipart/alternative is the only other type of multipart supported now.
+          */
+          multipart = [bodyStructureKey objectForKey: @"multipart"];
+          if ([multipart isEqualToString: @""])
+            mixedPart = NO;
+          else
+            {
```

Comment:
I like to put brackets if the statement spans more than one line, just for readability. If you feel symmetry is needed, I can add it to the positive statement.